etest_test/build_test.py

```diff
@@ -9,4 +14,193 @@ def test_help() -> None:
     """Ensure etest-build --help exits successfully."""
     runner = click.testing.CliRunner()
     result = runner.invoke(sut.main, ["--help"])
+
+    assert result.exit_code == 0
+
+
+def test_nobuild(mocker: MockerFixture) -> None:
+    """Ensure etest-build doesn't call any functions when not building."""
+    # Mock build functions
+    build = mocker.patch(
```

Comment:
I couldn't find a difference between the two systems, as the fixture is just a wrapper around `unittest.mock`. Im also not sure if the unittest mocks get reset between tests.

Reply:
Yes, but with the unittest interface we can decorate tests keeping the content of the test easier to follow.  Right now you have three different sections in the test rather than just two which makes it quite long winded at times.  That's why I was thinking using unittest mocks might be a tad clearer.  Both get reset between tests as the tests are single functions.  The context will go out of scope and cleanup will be done regardless of which you use.